spec.json

```diff
@@ -1,142 +1,161 @@
-      "description": ""
-    }
-  },
-  "schema": {
-    "0": {
-      "name": "Missing Headers",
-      "message": "One or more column headers is missing: {headers}",
-      "weight": 9,
-      "description": ""
-    },
-    "1": {
-      "name": "Non-Matching Headers",
-      "message": "The given headers do not match the schema: {schema_headers} vs. {given_headers}",
-      "weight": 9,
-      "description": ""
-    },
-    "2": {
-      "name": "Badly Ordered Headers",
-      "message": "The headers are badly ordered according to the schema: {schema_headers} vs. {given_headers}",
-      "weight": 5,
-      "description": ""
-    },
-    "3": {
-      "name": "Type Violation",
-      "message": "The value {value} in column {column} is not a valid {type}",
-      "weight": 9,
-      "description": ""
-    },
-    "4": {
-      "name": "Format Violation",
-      "message": "The value {value} in column {column} is not a valid {format}",
-      "weight": 7,
-      "description": ""
-    },
-    "5": {
-      "name": "Required Constraint",
-      "message": "Column {column} is a required field, but row {row} has no value",
-      "weight": 9,
-      "description": ""
-    },
-    "6": {
-      "name": "Unique Constraint",
-      "message": "Column {column} is a unique field, yet the value {value} found in row {row} has already been seen",
-      "weight": 9,
-      "description": ""
-    },
-    "7": {
-      "name": "Minimum Length Constraint",
-      "message": "The value {value} in column {column} does not conform to the minimum length constraint of {constraint}",
-      "weight": 7,
-      "description": ""
-    },
-    "8": {
-      "name": "Maximum Length Constraint",
-      "message": "The value {value} in column {column} does not conform to the maximum length constraint of {constraint}",
-      "weight": 7,
-      "description": ""
-    },
-    "9": {
-      "name": "Pattern Constraint",
-      "message": "The value {value} in column {column} does not conform to the pattern constraint of {constraint}",
-      "weight": 7,
-      "description": ""
-    },
-    "10": {
-      "name": "Minimum Value Constraint",
-      "message": "The value {value} in column {column} does not conform to the minimum value constraint of {constraint}",
-      "weight": 7,
-      "description": ""
-    },
-    "11": {
-      "name": "Maximum Value Constraint",
-      "message": "The value {value} in column {column} does not conform to the maximum value constraint of {constraint}",
-      "weight": 7,
-      "description": ""
-    },
-    "12": {
-      "name": "Enumerable Value Constraint",
-      "message": "The value {value} in column {column} does not conform to the given enumeration: {constraint}",
-      "weight": 7,
-      "description": ""
-    }
-  }
+    "version": "1.0.0-alpha1",
+    "errors": [
+        {
+          "code": "io-error",
+          "type": "general",
+          "context": "any",
+          "description": ""
+        },
+        {
```

Comment:
We are missing two things here that to me seem so critical, I'm not sure why they are not here:

1. `name`: we need a human readable name. GoodTables errors are collated into human-targeted reports. These currently manifest in the GoodTables web UI, and some client libs like in OpenSpending and DataPackagist. Why have you removed name?
2. `message`: like name, we need really clear human readable messages, that can be used by non-technical people to fix errors in their tabular data. Again, the GoodTables UI has many examples of this. In the original spec here, I presented a message string for each error than can easily be manipulated in any programming language that might implement this spec. Why remove it?